# Client Olist - Phase 1 Analyse du fichier source `Order Reviews`

## Importation des fichiers

In [22]:
import pandas as pd
import numpy as np

In [23]:
df_order_reviews = pd.read_csv('../SOURCE/olist_order_reviews_dataset.csv')

# Nombre de ligne et de colonne
print('Data order reviews', df_order_reviews.shape)

Data order reviews (99224, 7)


In [24]:
df_order_reviews.head()

review_id                          order_id  \
0  7bc2406110b926393aa56f80a40eba40  73fc7af87114b39712e6da79b0a377eb   
1  80e641a11e56f04c1ad469d5645fdfde  a548910a1c6147796b98fdf73dbeba33   
2  228ce5500dc1d8e020d8d1322874b6f0  f9e4b658b201a9f2ecdecbb34bed034b   
3  e64fb393e7b32834bb789ff8bb30750e  658677c97b385a9be170737859d3511b   
4  f7c4243c7fe1938f181bec41a392bdeb  8e6bfb81e283fa7e4f11123a3fb894f1   

   review_score review_comment_title  \
0             4                  NaN   
1             5                  NaN   
2             5                  NaN   
3             5                  NaN   
4             5                  NaN   

                              review_comment_message review_creation_date  \
0                                                NaN  2018-01-18 00:00:00   
1                                                NaN  2018-03-10 00:00:00   
2                                                NaN  2018-02-17 00:00:00   
3              Recebi bem antes do prazo estipulado.  2017-04-21 00:00:00   
4  Parabéns lojas lannister adorei comprar pela I...  2018-03-01 00:00:00   

  review_answer_timestamp  
0     2018-01-18 21:46:59  
1     2018-03-11 03:05:13  
2     2018-02-18 14:36:24  
3     2017-04-21 22:02:06  
4     2018-03-02 10:26:53

In [25]:
df_order_reviews.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 99224 entries, 0 to 99223
Data columns (total 7 columns):
 #   Column                   Non-Null Count  Dtype 
---  ------                   --------------  ----- 
 0   review_id                99224 non-null  object
 1   order_id                 99224 non-null  object
 2   review_score             99224 non-null  int64 
 3   review_comment_title     11568 non-null  object
 4   review_comment_message   40977 non-null  object
 5   review_creation_date     99224 non-null  object
 6   review_answer_timestamp  99224 non-null  object
dtypes: int64(1), object(6)
memory usage: 5.3+ MB


In [26]:
df_order_reviews.describe()

review_score
count  99224.000000
mean       4.086421
std        1.347579
min        1.000000
25%        4.000000
50%        5.000000
75%        5.000000
max        5.000000

In [27]:
df_order_reviews.isnull().any()

review_id                  False
order_id                   False
review_score               False
review_comment_title        True
review_comment_message      True
review_creation_date       False
review_answer_timestamp    False
dtype: bool

In [28]:
df_order_reviews.isna().sum()

review_id                      0
order_id                       0
review_score                   0
review_comment_title       87656
review_comment_message     58247
review_creation_date           0
review_answer_timestamp        0
dtype: int64

`Constat sur 99225 commentaire, il y a 87656 qui ne mettent pas de titre à leur commentaire et 58247 sans commentaire`

In [29]:
SansTitre = df_order_reviews['review_comment_title'].isna().sum() / df_order_reviews.shape[0] * 100
print(f"il y a {SansTitre}% qui ne mettre pas de titre à leur commentaire")

SansCommentaire = df_order_reviews['review_comment_message'].isna().sum() / df_order_reviews.shape[0] * 100
print(f"il y a {SansCommentaire}% qui ne mettre pas de commentaire")




il y a 88.34153027493349% qui ne mettre pas de titre à leur commentaire
il y a 58.70253164556962% qui ne mettre pas de commentaire


In [39]:
for col in df_order_reviews.columns:
    Nb_unique = df_order_reviews[col].unique().shape[0]
    Nb_value = df_order_reviews[col].count()
    print(f'dans la colonne {col} il y a {Nb_unique} unique et tant {Nb_value} de valeur')

dans la colonne review_id il y a 98410 unique et tant 99224 de valeur
dans la colonne order_id il y a 98673 unique et tant 99224 de valeur
dans la colonne review_score il y a 5 unique et tant 99224 de valeur
dans la colonne review_comment_title il y a 4528 unique et tant 11568 de valeur
dans la colonne review_comment_message il y a 36160 unique et tant 40977 de valeur
dans la colonne review_creation_date il y a 636 unique et tant 99224 de valeur
dans la colonne review_answer_timestamp il y a 98248 unique et tant 99224 de valeur
